mesonbuild/mesonmain.py

```diff
@@ -287,6 +288,15 @@ def run(original_args, mainfile):
         print('You have python %s.' % sys.version)
         print('Please update your environment')
         return 1
+    # https://github.com/mesonbuild/meson/issues/3653
+    if sys.platform.lower() == 'msys':
+        mlog.error('This python3 seems to be msys/python on MSYS2 Windows, which is known to have semantics incompatible with Meson')
+        msys2_arch = detect_msys2_arch()
+        if msys2_arch:
+            mlog.error('Please install and use mingw-w64-i686-python3 and/or mingw-w64-x86_64-python3 with Pacman')
+        else:
+            mlog.error('Please use download and use Python from www.python.org')
```

Comment:
I'm unclear as to when this 2nd case happens?
This is a msys/python, but MSYSTEM_CARCH isn't set so we recommend using a Wiindows python?

Reply:
This could happen if someone is using an MSYS python but not from inside an MSYS2 environment. In that case, we have no idea what is happening, but at least we can recommend a solution.

Reply:
I'm not sure how often that's going to be the right remedial advice, and how often it's just going to confuse the situation further

Reply:
Well, the native win32 python has always worked correctly for me in all environments (cmd, msys, msys2, powershell, double-click) if you use the right options, and hopefully with the screenshots it will be clearer. I guess I should link to `Getting-meson.html` here.